pkg/cmd/roachtest/operations/backup_restore.go

```diff
@@ -0,0 +1,140 @@
+package operations
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/cluster"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/operation"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/option"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/registry"
+	"github.com/cockroachdb/cockroach/pkg/cmd/roachtest/roachtestflags"
+	"github.com/cockroachdb/cockroach/pkg/testutils"
+	"github.com/cockroachdb/cockroach/pkg/util/randutil"
+)
+
+type backupRestoreCleanup struct {
+	db string
+}
+
+func (cl *backupRestoreCleanup) Cleanup(
+	ctx context.Context, o operation.Operation, c cluster.Cluster,
+) {
+	conn := c.Conn(ctx, o.L(), 1, option.VirtualClusterName(roachtestflags.VirtualCluster))
+	defer conn.Close()
+
+	o.Status(fmt.Sprintf("dropping newly created db %s", cl.db))
+	_, err := conn.ExecContext(ctx, fmt.Sprintf("DROP DATABASE %s CASCADE", cl.db))
+	if err != nil {
+		o.Fatal(err)
+	}
+}
+
+func runBackupRestore(
+	ctx context.Context, o operation.Operation, c cluster.Cluster,
+) registry.OperationCleanup {
+	// This operation looks for the district table in a database named cct_tpcc or tpcc.
+	rng, _ := randutil.NewPseudoRand()
+	dbWhitelist := []string{"cct_tpcc", "tpcc"}
+	conn := c.Conn(ctx, o.L(), 1, option.VirtualClusterName(roachtestflags.VirtualCluster))
+	defer conn.Close()
+	dbs, err := conn.QueryContext(ctx, "SELECT database_name FROM [SHOW DATABASES]")
+	if err != nil {
+		o.Fatal(err)
+	}
+	var dbName string
+	for dbs.Next() {
+		var dbStr string
+		if err := dbs.Scan(&dbStr); err != nil {
+			o.Fatal(err)
+		}
+		for i := range dbWhitelist {
+			if dbWhitelist[i] == dbStr {
+				// We found a db in the whitelist.
+				dbName = dbStr
+				break
+			}
+		}
+		if dbName != "" {
+			break
+		}
+	}
+	if dbName == "" {
+		o.Status("did not find a db in the whitelist")
+		return nil
+	}
+
+	backupTS := time.Now().Add(-1 * time.Minute).UTC().Format(time.DateTime)
+
+	o.Status(fmt.Sprintf("backing up table district in db %s", dbName))
+	bucket := fmt.Sprintf("gs://%s/operation-backup-restore/%d/?AUTH=implicit", testutils.BackupTestingBucket(), rng.Int63())
+
+	_, err = conn.ExecContext(ctx, fmt.Sprintf("BACKUP TABLE %s.district TO '%s' AS OF SYSTEM TIME '%s'", dbName, bucket, backupTS))
+	if err != nil {
+		o.Fatal(err)
+	}
+
+	restoreDBName := fmt.Sprintf("backup_restore_op_%d", rng.Int63())
+	_, err = conn.ExecContext(ctx, fmt.Sprintf("CREATE DATABASE %s", restoreDBName))
+	if err != nil {
+		o.Fatal(err)
+	}
+
+	o.Status(fmt.Sprintf("restoring table district into db %s", dbName))
+	_, err = conn.ExecContext(ctx, fmt.Sprintf("RESTORE TABLE %s.district FROM '%s' AS OF SYSTEM TIME '%s' WITH OPTIONS (into_db = '%s', skip_missing_foreign_keys)", dbName, bucket, backupTS, restoreDBName))
+
+	if err != nil {
+		o.Fatal(err)
+	}
+
+	o.Status(fmt.Sprintf("verifying table district in db %s", restoreDBName))
+	// We sum the d_next_o_id as a quick way to fingerprint the table, as the values
```

Comment:
Makes sense, just for my own curiosity since I'm not as familiar with the built-in fingerprints, is there a specific reason we didn't use the built-in fingerprint functionality [1]?
[1] https://github.com/cockroachdb/cockroach/blob/master/pkg/sql/show_fingerprints_test.go#L131

Reply:
There was no reason other than the fact that I'd never used our fingerprint functionality 😅 done.